### PR TITLE
Issue#96235 Add configuration changes needed for Power of Attorney request migrations

### PR DESCRIPTION
**Pull request by ojbucao**
This PR introduces necessary configuration updates to prepare the accredited_representative_portal module for upcoming migrations related to Power of Attorney Requests. These changes are prerequisites to ensure smooth integration and proper handling of the new database structures.

Key Changes:

1. Dependency Updates:
  • Added the blind_index gem to the Gemfile and gemspec to enable encrypted and hashed fields for secure data storage.
2. Engine Configuration:
  • Updated engine.rb to append migration paths, ensuring that app-wide migration commands in the parent application detect the engine’s migrations.
3. Lockfile Update:
  • Updated Gemfile.lock to reflect the addition of blind_index and ensure consistency across environments.

Justification:

These changes are required to support upcoming migrations, which rely on:
• The blind_index gem for encrypted fields with blind indexing.
• Proper engine migration path detection to enable database changes in the parent monolith.

By isolating these configuration changes, this PR ensures that the environment is correctly set up before introducing the database migrations in a subsequent PR.

Follow-Up:

A [separate PR ](https://github.com/department-of-veterans-affairs/vets-api/pull/19880)will include the actual migrations and schema.rb changes to adhere to the “migrations-only” rule enforced by the CI bot.

### CODE REVIEW

**Review by nihil2501 — CHANGES_REQUESTED** [review]
`table_name_prefix`?

File: modules/accredited_representative_portal/lib/accredited_representative_portal/engine.rb [hunk 1]
```diff
@@ -5,6 +5,16 @@ class Engine < ::Rails::Engine
     isolate_namespace AccreditedRepresentativePortal
     config.generators.api_only = true
```
[the remark this answers is not in the document]
Comment (ojbucao) [reply modules/accredited_representative_portal/lib/accredited_representative_portal/engine.rb:6]: Yeah I had thought about this. Felt like it belonged more with the models PR and not the migrations PR.

I can put it back.

[the remark this answers is not in the document]
Comment (nihil2501) [reply modules/accredited_representative_portal/lib/accredited_representative_portal/engine.rb:6]: oh i see what you mean. either way works then!

**Review by nihil2501 — APPROVED** [review]
LGTM

**Review by ericboehs — APPROVED** [review]

**Pull request merged**